### ISSUE TITLE
[bug] Meta logging is not reliable

### ISSUE THREAD

**Issue by lonix1** [winstonjs/winston-mongodb#156]
Adds to some of the concern [here](https://github.com/winstonjs/winston-mongodb/issues/150).

There is discrepancy between the readme (which should be "the truth"), the source, the issues, and the main library.

Two problems:
- There are various recommendations to use `metadata`, `meta`, `metaKey`, etc. Which is official?
- Most winston plugins log to the meta-within-meta format, but this package doesn't detect it. For example, the `express-winston` package logs meta to `... { "meta": { "metaKey": { ... } } }` or similar.

**Comment by lonix1** [comment]
I thought it was just me, but [it's not](https://github.com/winstonjs/winston/issues/1508). Meta isn't logged properly, **and it's not the fault of this package**.

But... that means the package should support the new format too.

Other winston plugins that produce logs don't get detected by this plugin. :(

**Comment by lonix1** [comment]
Found a hacky solution - during configuration, do this:

````
const mongoDbConnectionOptions: MongoDbConnectionOptions = {
  // config...
};
(mongoDbConnectionOptions as any).metaKey = "meta";     <--- magic
````

This hack is necessary because [the TypeScript definition file is missing some stuff](https://github.com/winstonjs/winston-mongodb/issues/155), including `metaKey`.

This makes it looks for "meta" (instead of "metadata"), which is used by winston and winston plugins.

One could say this is "by design", but it's a poor choice because the winston ecosystem is using "meta".

**Comment by pvanagtmaal** [comment]
@lonix1 I'm not entirely sure why you've closed this. It is technically a bug right? The least thing we can do is fix the type definitions.

**Comment by lonix1** [comment]
ok reopened

**Comment by kwadoskii** [comment]
Please how did you guys fix this issue on winston-mongodb version 5.0.5?
